design/hardware-classification-controller/support-for-new-parameters-hwcc.md

```diff
@@ -0,0 +1,192 @@
+   `status.hardwareDetails.cpu.arch` field for an exact string match.
+
+   User can provide `arch` type based on requirement.
+
+   Investigation Details: Below is sample introspection data for CPU.
+   Types of architecture are 32-bit (x86) and 64-bit
+   (x86-64, IA64, and AMD64).
+
+   ```yaml
+   hardware:
+       cpu:
+          arch: "x86_64"
+   ```
+
+1. `hardwareCharacteristics.firmware.bios`
+
+   Firmware is new parameter to be introduced in HardwareClassification
+   which contains BIOS information. User can specify BIOS requirements
+   by providing `hardwareCharacteristics.firmware.bios.vendor` and
+   `hardwareCharacteristics.firmware.bios.version`.
+
+   The classification of `vendor` will be done using exact string
+   match of the value provided by user in profile.
+
+   The classification of `version` will be done by checking if host
+   `version` is greater than equal to `minorVersion` and less than
+   equal to `majorVersion`. `minorVersion` and `majorVersion` are
+   provided by user in profile.
+
+   Investigation details: Below is sample introspection data for firmware.
+   `BIOS` field under firmware have two fields, `vendor` and `version`.
+
+    ```yaml
+    hardware:
+        firmware:
+             bios:
+               date: "10/17/2018"
+               vendor: "Dell Inc."
+               version: "1.5.6"
+    ```
+
+1. `hardwareCharacteristics.systemVendor`
+
+   SystemVendor is new parameter to be introduced which helps user to
+   classify hosts of certain manufacturer and product. User needs to
+   provide `hardwareCharacteristics.systemVendor.manufacturer` and
+   `hardwareCharacteristics.systemVendor.productName` in yaml.
+
+   The classification of `manufacturer` will be done using exact
+   string match of the values provided by user in profile.
+
+   `productName` from hardware details comes with SKU code and
+   in YAML user will provide only name of product.
+   The classification of `productName` will be done using sub
+   string match of the values provided by user in profile.
+
+   Investigation details: Below is sample introspection data for
+   systemVendor. `systemVendor` field contains `manufacturer` and
+   `productName`.
+
+    ```yaml
+    hardware:
+        systemVendor:
+          manufacturer: Dell Inc.
+          productName: PowerEdge R640 (SKU=NotProvided;ModelName=PowerEdge R640)
+    ```
+
+### Implementation Details/Notes/Constraints
+
+Link for Existing HWCC Specs
+[Existing YAML](https://github.com/metal3-io/hardware-classification-controller/blob/master/config/samples/metal3.io_v1alpha1_hardwareclassification.yaml)
+
+* Below is sample yaml for additional parameters in HardwareClassification.
+
+   ```yaml
+
+   apiVersion: metal3.io.sigs.k8s.io/v1alpha1
+   kind: HardwareClassification
+   metadata:
+     name: profile1
+   spec:
+     hardwareCharacteristics:
+       CPU:
+          arch: x86_64
+          minimumCount: 4
+          minimumSpeed: 4.3
```

Comment:
This needs units

Reply:
Units will be updated in User Guide under [User guide](https://github.com/metal3-io/hardware-classification-controller/blob/master/docs/user-guide.md)

Reply:
It would be better if the field names included the units, so that users don't have to guess or remember or look it up every time they want to use it. We also don't want `float` fields because some values don't pass cleanly from text to `float64`, so it'll need to be `minimumSpeedMHz` defined as an `int` or `int32` and the value in the example would become `4300`.

Reply:
Yes, we will append units as per the field names.